src/super_gradients/training/datasets/detection_datasets/pascal_voc_format_detection.py

```diff
@@ -0,0 +1,198 @@
+        │   │       ├──ImageSets/Action
+        │   │       ├──ImageSets/Layout
+        │   │       ├──Annotations
+        │   │       └──SegmentationObject
+        │   ├─train2007
+        │   ├─test2007
+        │   └─val2007
+        └─labels
+            ├─train2012
+            ├─val2012
+            ├─train2007
+            ├─test2007
+            └─val2007
+
+    Note:
+        If both 'images_sub_directory' and ('images_dir', 'labels_dir') are provided, a warning will be raised.
+
+    Usage:
+        voc_2012_train = PascalVOCDetectionDataset(data_dir="./data/pascal_voc",
+                                            images_dir="images/train2012/JPEGImages",
+                                            labels_dir="labels/train2012/Annotations",
+                                            download=True)
+    """
+
+    @deprecated_parameter(
+        "images_sub_directory",
+        deprecated_since="3.7.0",
+        removed_from="3.8.0",
+        reason="Support of `images_sub_directory` is removed since it allows less flexibility." " Please use 'images_dir' and 'labels_dir' instead.",
+    )
+    def __init__(
+        self,
+        data_dir: str,
+        images_dir: str,
+        labels_dir: str,
+        max_num_samples: int = None,
+        cache_annotations: bool = True,
+        input_dim: Union[int, Tuple[int, int], None] = None,
+        transforms: List[AbstractDetectionTransform] = [],
+        all_classes_list: Optional[List[str]] = [],
+        class_inclusion_list: Optional[List[str]] = None,
+        ignore_empty_annotations: bool = True,
+        verbose: bool = True,
+        show_all_warnings: bool = False,
+        cache=None,
+        cache_dir=None,
+    ):
+        """
+        Initialize the Pascal VOC Detection Dataset.
+
+        """
+
+        self.data_dir = data_dir
+
+        self.images_dir = os.path.join(data_dir, images_dir)
+        self.labels_dir = os.path.join(data_dir, labels_dir)
+
+        super(PascalVOCFormatDetectionDataset, self).__init__(
+            data_dir=data_dir,
+            original_target_format=XYXY_LABEL,
+            max_num_samples=max_num_samples,
+            cache_annotations=cache_annotations,
+            input_dim=input_dim,
+            transforms=transforms,
+            all_classes_list=all_classes_list,
+            class_inclusion_list=class_inclusion_list,
+            ignore_empty_annotations=ignore_empty_annotations,
+            verbose=verbose,
+            show_all_warnings=show_all_warnings,
+            cache=cache,
+            cache_dir=cache_dir,
+        )
+
+    def _setup_data_source(self) -> int:
+        """Initialize img_and_target_path_list and warn if label file is missing
+
+        :return: List of tuples made of (img_path,target_path)
+        """
+        if not Path(self.images_dir).exists():
+            raise FileNotFoundError(f"{self.images_dir} not found.")
+
+        img_files = glob.glob(os.path.join(self.images_dir, "*.jpg"))
```

Comment:
Pls add list(sorted(...)) 
Why it is important:
glob return images in no particular order:
> Return a list containing the names of the entries in the directory given by path. The list is in arbitrary order.

So it may happen that in DDP you would get different orders, and thus some images would be excluded from train/validation and some appear more than onces

Reply:
Nice catch!
done in latest commit.